### PR TITLE
feat: ADR-008: Fees module

### PR DESCRIPTION
**Pull request by bragaz**
## Description

Closes: #XXXX.

This PR implements the `x/fees` module that allows governance to set minimum fees on messages.

Depends-On: #816
Depends-On: #817

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [x] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [x] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by RiccardoM — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/cosmos/cosmos-sdk v0.45.1
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/cosmos/ibc-go/v2 v2.2.0
-	github.com/ethereum/go-ethereum v1.10.17
+	github.com/ethereum/go-ethereum v1.10.16
```
Comment (dadamu) [inline go.mod:12]:
```suggestion
	github.com/ethereum/go-ethereum v1.10.17
```
It should be updated to v1.10.17 already by #799.

Comment (RiccardoM) [reply go.mod:12]: The problem is this `go-ethereum` version breaks things with `btced` which leads to `go mod tidy` not working properly for systems that have a Go version prior to 1.17

Comment (RiccardoM) [reply go.mod:12]: I've opened #816 to make this clear

File: x/fees/keeper/fee_decorator_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+
+	"github.com/desmos-labs/desmos/v3/x/fees/keeper"
+	"github.com/desmos-labs/desmos/v3/x/fees/types"
+	profilestypes "github.com/desmos-labs/desmos/v3/x/profiles/types"
+)
+
+func (suite *KeeperTestSuite) TestFeeDecorator() {
```
Comment (dadamu) [inline x/fees/keeper/fee_decorator_test.go:12]: Can you add a test which is when BlockHeight == 0?

Comment (bragaz) [reply x/fees/keeper/fee_decorator_test.go:12]: @dadamu added.

Comment (bragaz) [reply x/fees/keeper/fee_decorator_test.go:12]: I can't re-request you the review. I guess it's a GitHub bug.

File: x/fees/keeper/fee_decorator_test.go [hunk 3]
```diff
@@ -67,18 +95,23 @@ func (suite *KeeperTestSuite) TestFeeDecorator() {
...
+			simulate:   true,
 		},
 	}
 
 	for _, tc := range testsCases {
 		ctx, _ := suite.ctx.CacheContext()
 		if tc.store != nil {
+			if tc.zeroHeight {
+				ctx = suite.ctx.WithBlockHeight(0)
+			}
```
Comment (RiccardoM) [inline x/fees/keeper/fee_decorator_test.go:109]: This is not how we should handle things here, as it makes things complicated. Instead what we should do is have a new optional field inside the `testCases` structure: 
```go
malleate func(ctx sdk.Context) sdk.Context
```

And then use it as follows:
```go
ctx, _ := suite.ctx.CacheContext()
if tc.malleate != nil {
  ctx = tc.malleate(ctx)
}
if tc.store != nil {
  tc.store(ctx)
}
```

**Review by RiccardoM — CHANGES_REQUESTED** [review]

File: x/fees/keeper/fee_decorator_test.go (hunk 2 above)
Comment (dadamu) [reply x/fees/keeper/fee_decorator_test.go:12]: I can't  review as well. lol

Comment (dadamu) [reply x/fees/keeper/fee_decorator_test.go:12]: Never mind it is back.

**Review by dadamu — APPROVED** [review]
Ready to go to me.

**Review by RiccardoM — APPROVED** [review]

**Pull request merged**